governance/default.go

```diff
@@ -665,6 +666,14 @@ func getGovernanceCacheKey(num uint64) common.GovernanceCacheKey {
 
 // Store new governance data on DB. This updates Governance cache too.
 func (g *Governance) WriteGovernance(num uint64, data GovernanceSet, delta GovernanceSet) error {
+	g.idxCacheLock.RLock()
+	idx := make([]uint64, 0, len(g.idxCache))
+	copy(idx, g.idxCache)
+	g.idxCacheLock.RUnlock()
+
+	if len(idx) > 0 && num <= idx[len(idx)-1] {
```

Comment:
Same routine is in `addGovernanceCache`. 
Is it duplicated? If so, we can refactor them?

Reply:
@KimKyungup That's a good point. Thanks! I added a new commit for this. Please take a look.

Reply:
@KimKyungup I rolled back the last commit. As @ehnuje pointed out, the behavior of `addGovernanceCache` is different from that of `WriteGovernance`, the last commit didn't work.